nujo/autodiff/_functions/_transform.py

```diff
@@ -53,3 +55,134 @@ def backward(self, idx: int, accum_grad: Function.T) -> Function.T:
 
 
 # ====================================================================================================
+
+
+class _Pad(Function):
+    ''' Zero Padding
+
+    Pads an image shaped array with zeros.
+
+    '''
+    def __init__(self, input: Union[Tensor, ndarray, List[Number], Number],
+                 padding: Tuple[int, int]):
+
+        super(_Pad, self).__init__(input)
+
+        # Shape of `input` should be: (batch_size, channels, height, width)
+        assert len(self.children[0].shape) == 4
+
+        self.padding = padding
+
+    def forward(self) -> ndarray:
+        return pad(self.children[0].value, (
+            (0, 0),
+            (0, 0),
+            (self.padding[0], self.padding[0]),
+            (self.padding[1], self.padding[1]),
+        ))
```

Comment:
I looked up what pad does but couldn't get what this is trying to do

Reply:
You have a padding `(before, after)` for each dimension `(batch_size, channels, height, width)` and since you only need to pad *height and width* you write `(0, 0)` for the rest dimensions